src/exe/Main.hs

```diff
@@ -125,7 +134,21 @@ optionDescriptors =
     (  "Optional. Path to output directory.\n"
     ++ "Prints to the console by default."
     )
+  , Option
+    ['f']
+    ["frontend"]
+    (ReqArg (\f opts -> opts { optFrontend = parseFrontend f }) "FRONTEND")
+    (  "Optional. Frontend to use for input parsing.\n"
+    ++ "Uses haskell-src-extensions by default"
+    )
   ]
+ where
+  -- | Parses a given string to one of the frontents. Returns @HSE@ by
+  --   default.
+  parseFrontend :: String -> Frontend
+  parseFrontend s = if map toLower s == "ghclib" || map toLower s == "ghc-lib"
+    then GHClib
+    else HSE
```

Comment:
I suggest also comparing the string to "hse" and "haskell-src-exts" and printing an error message if an unavailable front end has been selected.

Reply:
:+1: for the suggestion of printing errors for unsupported front ends.